packages/components/organisms/f-checkout/src/components/Selector.vue

```diff
@@ -40,36 +40,48 @@ export default {
         * Returns an array of formatted dates to display as options in a dropdown
         */
         fulfilmentTimes () {
-            const times = this.availableFulfilment.isAsapAvailable
-                ? [this.$t('asapFulfilmentOption')]
-                : [];
-
+            const times = [];
             this.availableFulfilment.times.forEach(time => {
                 const formattedTime = this.$d(new Date(time.from), 'short');
 
-                times.push(formattedTime);
+                times.push({
+                    text: formattedTime,
+                    value: time.from
+                });
             });
 
+            if (this.availableFulfilment.isAsapAvailable && times.length) {
+                times[0].text = this.$t('asapFulfilmentOption');
+            }
+
             return times;
         }
     },
 
     watch: {
         'fulfilmentTimes' () {
             if (this.fulfilmentTimes.length > 0) {
-                this.selectionChanged(this.fulfilmentTimes[0]);
+                this.selectionChanged(this.fulfilmentTimes[0].value);
             }
         }
     },
 
     methods: {
+        ...mapActions('checkout', [
+            'updateFulfilmentTime'
+        ]),
+
         /**
         * Update the selected available fulfilment time.
         *
         * @param {string} selectedFulfilmentTime The time emited when dropdown value is changed.
         **/
         selectionChanged (selectedFulfilmentTime) {
             this.selectedAvailableFulfilmentTime = selectedFulfilmentTime;
+            this.updateFulfilmentTime({
+                from: selectedFulfilmentTime,
+                to: selectedFulfilmentTime
```

Comment:
I understand the APIs will support different from/to times at a later date? We can update this here when that time comes.

Reply:
Is it worth adding a comment here about here. It's hard to see that the `selectedFulfilmentTime` differs on glance.